File: 프로그래머스/0/120909. 제곱수 판별하기/제곱수 판별하기.java
class Solution {
    public int solution(int n) {
        int sqrt = (int) Math.sqrt(n);
        int answer = 0;
        
        if ( sqrt * sqrt == n) {
            answer = 1;
        }
        else {
            answer = 2;
        }
        
        return answer;
    }
}

File: 프로그래머스/0/120909. 제곱수 판별하기/제곱수 판별하기.c
#include <stdio.h>
#include <math.h>

int solution(int n) {
    int sqrt_n = (int)sqrt(n);
    
    if (sqrt_n * sqrt_n == n) {
        return 1;
    } else {
        return 2;
    }
}

int main() {
    int n;
    scanf("%d", &n);
    printf("%d\n", solution(n));
    return 0;
}
